### PR TITLE
Fix_issue_2239: Nasa Exoplanet Archive query_object method suddenly produces InvalidTableError

### PR DESCRIPTION
**Pull request by rickynilsson**
Fix query_aliases(). Now using `aliaslookup` service instead of deprecated `aliastable`.

### CODE REVIEW

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py [hunk 1]
```diff
@@ -370,14 +372,14 @@ def query_object_async(self, object_name, *, table="ps", get_query_payload=False
         return self.query_criteria_async(table, get_query_payload=get_query_payload, cache=cache, **criteria)
 
     @class_or_instance
-    def query_aliases(self, object_name, *, cache=None):
+    def query_aliases(self, object_name):
```
Comment (bsipocz) [inline astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py:375]: why remove `cache`? If remove, clear out the docstring, too

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py [hunk 2]
```diff
@@ -387,18 +389,36 @@ def query_aliases(self, object_name, *, cache=None):
         response : list
             A list of aliases found for the object name. The default name will be listed first.
         """
-        return list(
-            self.query_criteria(
-                "aliastable", objname=object_name.strip(), cache=cache, format="csv"
-            )["aliasdis"]
-        )
+        url = requests.get("https://exoplanetarchive.ipac.caltech.edu/cgi-bin/Lookup/nph-aliaslookup.py?objname="+object_name)
```
Comment (bsipocz) [inline astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py:392]: I think this messes with the assumption in the mock test that expects an URL derived from the config, maybe these lines need to be revised when testing the alias resolution:

```
def mock_get(self, method, url, *args, **kwargs):  # pragma: nocover
    assert url == conf.url_api
 ```

**Review by bsipocz — COMMENTED** [review]
I'm pretty sure that the test failure is coming from the URL assumption in the mock setup, if you test it locally it should fail there, too, not just for old versions.

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py (hunk 1 above)
Comment (rickynilsson) [reply astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py:375]: Wasn't thinking, will add it back.

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py [hunk 3]
```diff
@@ -387,18 +391,36 @@ def query_aliases(self, object_name, *, cache=None):
...
```
Comment (bsipocz) [inline astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py:394]: As you added the base URL for the lookup to the config, please us it from there.

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py [hunk 4]
```diff
@@ -119,16 +119,16 @@ def patch_get(request):  # pragma: nocover
     return mp
 
 
-def test_regularize_object_name(patch_get):
-    NasaExoplanetArchiveMock = NasaExoplanetArchiveClass()
+# def test_regularize_object_name(patch_get):
```
Comment (bsipocz) [inline astroquery/ipac/nexsci/nasa_exoplanet_archive/tests/test_nasa_exoplanet_archive.py:122]: Please either uncomment or if the `regularize` keyword is removed, then remove this test, too.

File: astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py [hunk 5]
```diff
@@ -665,22 +687,18 @@ def query_planet(self, planet_name, cache=None, regularize=True, **criteria):
...
-        if regularize:
-            planet_name = self._regularize_object_name(planet_name)
         criteria = self._handle_all_columns_argument(**criteria)
         criteria["where"] = "pl_name='{0}'".format(planet_name.strip())
         return self.query_criteria("exoplanets", cache=cache, **criteria)
 
     @deprecated(since="v0.4.1", alternative="query_object")
     @deprecated_renamed_argument(["show_progress", "table_path"],
                                  [None, None], "v0.4.1", arg_in_kwargs=True)
-    def query_star(self, host_name, cache=None, regularize=True, **criteria):
```
Comment (bsipocz) [inline astroquery/ipac/nexsci/nasa_exoplanet_archive/core.py:683]: This API change needs to be mentioned in the changelog

**Review by bsipocz — COMMENTED** [review]
This fix needs to be mentioned in the changelog as the bug was reported by users. (And also add a mention of the API change for the removed keyword).

Other than the minor comments I left in the review, there is one remote test failing, maybe we could test not for the exact number of aliases but that there are e.g. more than 10? 
And if you run the remote test, they would also reveal some docs example inconsistencies. For some of those, e.g. where we expect changes as the upstream database grows, I would suggest to use `# doctest: +IGNORE_OUTPUT`.

**Comment by rickynilsson** [comment]
@bsipocz - I fixed up the testing and other things you mentioned, but seems codecov isn't happy. Any suggested tests I could add? I actually am not quite sure why the coverage isn't higher -- the tests should be running most of the added code.

**Comment by bsipocz** [comment]
@rickynilsson - I'll need to look into this more carefully to suggest actual code changes, but it seems that the mock testing is not actually running anything from the real method from the module, but only the mock function defined in the test, and that triggers the decrease in the coverage number.

**Comment by bsipocz** [comment]
@rickynilsson - I'll need to look into this more carefully to suggest actual code changes, but at first glance, it seems that the mock testing is not actually running anything from the real method from the module, but only the mock function defined in the test, and that triggers the decrease in the coverage number.

**Comment by bsipocz** [comment]
@rickynilsson - do you plan to add some extra test coverage here as part of the PR, or we should rather go ahead and merge and add more tests in a follow-up?

Before merging: doc examples should be fixed (if you run `python setup.py -P ipac.nexsci -R` it picks up the docs file, too), and a rebase should fix the changelog conflict.

**Comment by rickynilsson** [comment]
I should probably just make a final push on this to get the necessary mock tests in place. I'll set off some time to do this tomorrow.

**Comment by rickynilsson** [comment]
@bsipocz - something is obviously wrong in the mock test I added but I can't figure out exactly what. Any hints? The monkey patching stuff is a bit confusing to me.

**Review by bsipocz — APPROVED** [review]
There is a super nice 92% coverage, with and another nice 83% remote only coverage, so I'm fairly certain we got the majority of use cases under control :)

**Comment by bsipocz** [comment]
Thank you @rickynilsson!

**Pull request merged**